File: server/Models/Contract.cs
using Microsoft.EntityFrameworkCore;
using server.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RedHouse_Server.Models
{
    public class Contract
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(Offer))]
        public int OfferId { get; set; }
        public Offer? Offer { get; set; }
        public int? LawerId { get; set; }
        public User? Lawer { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string ContractType { get; set; }
        public string ContractStatus { get; set; }
        public double Earnings { get; set; }
        public int IsShouldPay { get; set; }
        public List<Milestone>? Milestones { get; set; }
        public List<ContractActivity>? ContractActivities { get; set; }


    }

}

File: mobile/lib/model/contract.dart
import 'package:client/model/milestone.dart';
import 'package:client/model/offer.dart';
import 'package:client/model/user.dart';
import 'package:json_annotation/json_annotation.dart';
part "contract.g.dart";

@JsonSerializable()
class Contract {
  int id;
  int offerId;
  DateTime startDate;
  DateTime? endDate;
  String contractType;
  String contractStatus;
  double earnings;
  int isShouldPay;
  Offer? offer;
  List<Milestone>? milestones;
  int? lawyerId;
  User? lawyer;

  Contract({
    required this.id,
    required this.offerId,
    required this.startDate,
    this.endDate,
    required this.contractType,
    required this.contractStatus,
    required this.earnings,
    required this.isShouldPay,
    this.offer,
    this.milestones,
    this.lawyerId,
    this.lawyer,
  });

  factory Contract.fromJson(Map<String, dynamic> json) =>
      _$ContractFromJson(json);
  Map<String, dynamic> toJson() => _$ContractToJson(this);
}
